pkg/agent/scalingevents/clients.go

```diff
@@ -0,0 +1,63 @@
+package scalingevents
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/lithammer/shortuuid"
+	"go.uber.org/zap"
+
+	"github.com/neondatabase/autoscaling/pkg/reporting"
+)
+
+type ClientsConfig struct {
+	S3 *S3ClientConfig `json:"s3"`
+}
+
+type S3ClientConfig struct {
+	reporting.BaseClientConfig
+	reporting.S3ClientConfig
+	PrefixInBucket string `json:"prefixInBucket"`
+}
+
+type eventsClient = reporting.Client[ScalingEvent]
+
+func createClients(ctx context.Context, logger *zap.Logger, cfg ClientsConfig) ([]eventsClient, error) {
+	var clients []eventsClient
+
+	if c := cfg.S3; c != nil {
+		generateKey := newBlobStorageKeyGenerator(c.PrefixInBucket)
+		client, err := reporting.NewS3Client(ctx, c.S3ClientConfig, generateKey)
+		if err != nil {
+			return nil, fmt.Errorf("error creating S3 client: %w", err)
+		}
+		logger.Info("Created S3 client for scaling events", zap.Any("config", c))
+
+		clients = append(clients, eventsClient{
+			Name:           "s3",
+			Base:           client,
+			BaseConfig:     c.BaseClientConfig,
+			SerializeBatch: reporting.WrapSerialize[ScalingEvent](reporting.GZIPCompress, reporting.JSONLinesMarshalBatch),
+		})
+	}
+
+	return clients, nil
+}
+
+// Returns a function to generate keys for the placement of scaling events data into blob storage.
+//
+// Example: prefix/2024/10/31/23/events_{uuid}.ndjson.gz (11pm on halloween, UTC)
+func newBlobStorageKeyGenerator(prefix string) func() string {
```

Comment:
Can we reuse the same key generator as we have for billing?

Reply:
We could, but we'd end up changing the format. I figured basically:

1. It's a little annoying to make the same key generator available to both packages; and
2. It's _probably_ better to use the same key format as what proxy provides, rather than matching what we use for billing

Not 100% sure though. WDYT?